src/app/data-model/DecodableList.h

```diff
@@ -102,7 +117,36 @@ class DecodableList
 
             if (mStatus == CHIP_NO_ERROR)
             {
-                mStatus = Decode(mReader, mValue);
+                mStatus = DataModel::Decode(mReader, mValue);
+            }
+
+            return (mStatus == CHIP_NO_ERROR);
+        }
+
+        template <typename X = T,
+                  typename std::enable_if_t<
+                      std::is_class<X>::value &&
+                          std::is_same<decltype(DataModel::Decode(std::declval<TLV::TLVReader &>(), std::declval<X &>(),
+                                                                  std::declval<Optional<FabricIndex> &>())),
+                                       CHIP_ERROR>::value,
+                      X> * = nullptr>
+        bool Next()
+        {
+            if (mReader.GetContainerType() == TLV::kTLVType_NotSpecified)
+            {
+                return false;
+            }
+
+            if (mStatus == CHIP_NO_ERROR)
+            {
+                mStatus = mReader.Next();
+            }
+
+            if (mStatus == CHIP_NO_ERROR)
+            {
+                mStatus = DataModel::Decode(mReader, mValue,
+                                            mFabricIndex == kUndefinedFabricIndex ? Optional<FabricIndex>::Missing()
+                                                                                  : Optional<FabricIndex>(mFabricIndex));
```

Comment:
I'm not sure I understand something, and this line highlights it.  We already have a way to represent "no fabric": it's `kUndefinedFabricIndex`.  Why do we need an `Optional<FabricIndex>` when calling Decode?